File: models/appUser.js
module.exports = (sequelize, DataTypes) => {
  const AppUser = sequelize.define('AppUser', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    username: {
      type: DataTypes.STRING(50),
      allowNull: false,
      unique: true
    },
    password: {
      type: DataTypes.STRING,
      allowNull: false
    },
    isActive: {
      type: DataTypes.BOOLEAN,
      defaultValue: true
    },
    lastLogin: {
      type: DataTypes.DATE,
      allowNull: true
    }
  }, {
    timestamps: true,
    tableName: 'app_users'
  });

  // Instance methods
  AppUser.prototype.toJSON = function() {
    const values = { ...this.dataValues };
    delete values.password;
    return values;
  };

  return AppUser;
};

File: backend_go/internal/models/app_user.go
package models

import "time"

// AppUser represents a mobile/app-facing user (separate from admin User).
// Sequelize model: AppUser — table: app_users
type AppUser struct {
	ID        uint       `gorm:"primaryKey;autoIncrement" json:"id"`
	Username  string     `gorm:"type:varchar(255);not null;uniqueIndex" json:"username"`
	Password  string     `gorm:"type:varchar(255);not null" json:"-"`
	IsActive  bool       `gorm:"default:true" json:"isActive"`
	LastLogin *time.Time `gorm:"index" json:"lastLogin"`
	CreatedAt time.Time  `json:"createdAt"`
	UpdatedAt time.Time  `json:"updatedAt"`

	// HasMany relationships — GORM uses slice fields to load associations.
	// These are NOT database columns; they're populated only when you use Preload().
	// Equivalent to Sequelize's hasMany(...) associations.
	Tokens           []AppUserToken    `gorm:"foreignKey:AppUserID" json:"-"`
	HouseholdMembers []HouseholdMember `gorm:"foreignKey:AppUserID" json:"-"`
	Expenses         []Expense         `gorm:"foreignKey:AppUserID" json:"-"`
	Incomes          []Income          `gorm:"foreignKey:AppUserID" json:"-"`
	Notes            []Note            `gorm:"foreignKey:AppUserID" json:"-"`
}

func (AppUser) TableName() string { return "app_users" }
